michbl/assignment4/assignment4.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include "codeu_test_lib.h"
+
+#define LAND true
+#define SEA false
```

Comment:
You can define global constants instead of using the preprocessor's `#define`:
```C++
const bool LAND = true;
const bool SEA = false;
```
In C++ using constants is the recommended way to define global constant values. They can be scoped (defined in a namespace or a class, e,g, MyClass::LAND), and can give better compile time errors.

Another option is to use enum as you have used for the Marking.

If you want to keep it `bool` (for example if the interface must be a grid of bool), you can give the grid a name which better represents the meaning of its content, e.g. `isLand` instead of `grid`. This way the boolean expressions can be more meaningful:
```C++
if (isLand[row][col]) { /* Mark the land */ }
```